src/module/system/statistic/statistic.ts

```diff
@@ -307,26 +307,30 @@ class StatisticCheck<TParent extends Statistic = Statistic> {
         });
 
         // Acquire additional adjustments for check-only modifiers using the parent's domains
-        const checkOnlyModifiers = [
-            data.check?.modifiers ?? [],
-            extractModifiers(parent.actor.synthetics, data.check?.domains ?? []),
-        ]
-            .flat()
-            .map((modifier) => {
-                modifier.adjustments.push(
-                    ...extractModifierAdjustments(
-                        parent.actor.synthetics.modifierAdjustments,
-                        parent.domains,
-                        this.parent.slug
-                    )
-                );
-                return modifier;
-            });
+        const checkOnlyModifiers = R.compact(
+            [
+                data.check.modifiers?.map((modifier) => {
+                    modifier.adjustments.push(
+                        ...extractModifierAdjustments(
+                            parent.actor.synthetics.modifierAdjustments,
+                            parent.domains,
+                            this.parent.slug
+                        )
+                    );
+                    return modifier;
+                }),
+                extractModifiers(parent.actor.synthetics, data.check?.domains ?? []),
+            ].flat()
+        );
         const rollOptions = parent.createRollOptions(this.domains, config);
         this.modifiers = [
             ...parentModifiers,
             ...checkOnlyModifiers.map((modifier) => modifier.clone({ test: rollOptions })),
         ];
+        if (this.type === "flat-check" && this.modifiers.length > 0) {
+            throw ErrorPF2e("Flat checks cannot have modifiers");
```

Comment:
I'd filter out the modifiers and do a warning instead. Best not to crash actor data preparation because someone decided to get frisky.